packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -470,6 +471,33 @@ contract OptimismPortal_Test is CommonTest {
         optimismPortal.setGasPayingToken({ _token: _token, _decimals: _decimals, _name: _name, _symbol: _symbol });
     }
 
+    /// @notice Ensures that the deposit event is correct for the `setGasPayingToken`
+    ///         code path that manually emits a deposit transaction outside of the
+    ///         `depositTransaction` function. This is a simple differential test.
+    function test_setGasPayingToken_correctEvent_succeeds() external {
+        vm.recordLogs();
+
+        vm.prank(address(systemConfig));
+        optimismPortal.setGasPayingToken({ _token: address(0x9999), _decimals: 19, _name: "", _symbol: "" });
```

Comment:
Would it be better to use nonzero values for name and symbol, just to avoid any edge cases where something isn't being set somewhere, and also to exercise maximum gas expenditure (since we now know that can be an issue)?

Reply:
Moreover, I think this test could be made stronger by making it a fuzz test with symbolic values passed into `setGasPayingToken()`.

Reply:
I have turned this into a fuzz test, you are right @maurelian